src/components/TransactionsTracker/useClosureRef.ts

```diff
@@ -0,0 +1,11 @@
+import { useEffect, useRef } from 'react';
+
+export const useClosureRef = <T>(value: T) => {
+  const refVal = useRef<T>(value);
+
+  useEffect(() => {
+    refVal.current = value;
+  }, [value]);
+
+  return refVal;
```

Comment:
probably should return refVal.current, to avoid reading .current everywhere

Reply:
It would not work. I already tried this solution